python/ray/data/datasource/file_based_datasource.py

```diff
@@ -65,32 +67,58 @@ class FileShuffleConfig:
     """Configuration for file shuffling.
 
     This configuration object controls how files are shuffled while reading file-based
-    datasets.
+    datasets. The random seed behavior is determined by the combination of ``seed``
+    and ``reseed_after_epoch``:
+
+    - If ``seed`` is None, the random seed is always None (non-deterministic shuffling).
+    - If ``seed`` is not None and ``reseed_after_epoch`` is False, the random seed is
+      constantly ``seed`` across epochs.
+    - If ``seed`` is not None and ``reseed_after_epoch`` is True, the random seed is
+      ``seed + epoch_idx``.
 
     .. note::
         Even if you provided a seed, you might still observe a non-deterministic row
         order. This is because tasks are executed in parallel and their completion
         order might vary. If you need to preserve the order of rows, set
-        `DataContext.get_current().execution_options.preserve_order`.
+        ``DataContext.get_current().execution_options.preserve_order``.
 
     Args:
-        seed: An optional integer seed for the file shuffler. If provided, Ray Data
-            shuffles files deterministically based on this seed.
+        seed: An optional integer seed for the file shuffler. If None, shuffling is
+            non-deterministic. If provided, shuffling is deterministic based on this
+            seed and the ``reseed_after_epoch`` setting.
+        reseed_after_epoch: If True, the random seed is ``seed + epoch_idx``, resulting
```

Comment:
we can just say the epoch index will be factored in the seed, `seed + epoch_idx` is an implementation detail.